controllers/cluster_controller.go

```diff
@@ -189,7 +190,7 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 			gvk := child.GetObjectKind().GroupVersionKind().String()
 
 			logger.Info("Deleting child", "gvk", gvk, "name", accessor.GetName())
-			if err := r.Client.Delete(context.Background(), child); err != nil {
+			if err := r.Client.Delete(context.Background(), child, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
```

Comment:
I worry that foreground deletion here would lead to blocking other reconciliations. At current we should be requeuing after the background deletions here and then only continuing after all of the "child" Machines have been successfully deleted.

Reply:
We've had some conversation in #1643 - do you think it would make sense to continue it there?